app/views/assessments/_partial_feedback_js.html.erb

```diff
@@ -1,22 +1,26 @@
 <%= external_javascript_include_tag "lodash" %>
 <script>
+  function loadPartialFeedback(partialFeedback) {
+    $('#partial-feedback').html(`
+      <pre>${partialFeedback}</pre>
+      <div class="autograding-in-progress">
+        <i class="refresh-feedback small material-icons" onclick="rotateIcon(); onClickRefresh()">refresh</i>
+        <pre>Autograding in progress</pre>
+      </div>`);
+    $('.feedback-status').html("In-Progress");
+    $('.feedback-status').removeClass("feedback-status__queued");
+    $('.feedback-status').addClass("feedback-status__inprogress");
+    $( "#result-body" ).load(window.location.href + " #result-body" );
+  }
   function refreshPartial(clicked) {
+    const lastPartialFeedback = localStorage.getItem(<%= @job_id %>);
```

Comment:
job_id does not uniquely identify a job - this counter resets whenever Tango server resets. 
You probably need the submission number (and maybe the user).

Reply:
At the same time, I believe it isn't as appropriate to use localStorage for storing the feedback as the they do not expire (and won't be deleted automatically) So a students localstorage will get bloated after many submissions to autolab.

Either somehow set an expiry, or explore sessionStorage
https://www.digitalocean.com/community/tutorials/js-introduction-localstorage-sessionstorage

Reply:
job_id only uniquely identifies on the Tango side - basically there will not be two jobs running that has the same job_id. So you won't get a situation where you get someone else's partial feedback